app/graph/types/public_team_type.rb

```diff
@@ -1,28 +1,27 @@
 PublicTeamType = GraphqlCrudOperations.define_default_type do
   name 'PublicTeam'
-  description 'Public team type'
+  description 'The public attributes of a team.'
 
   interfaces [NodeIdentification.interface]
 
-  field :name, !types.String
-  field :slug, !types.String
-  field :description, types.String
-  field :dbid, types.Int
-  field :avatar, types.String
-  field :private, types.Boolean
-  field :team_graphql_id, types.String
+  field :name, !types.String, 'Name'
+  field :slug, !types.String, 'Slug (URL path)'
+  field :description, types.String, 'Description'
+  field :avatar, types.String, 'Picture' # TODO Rename to 'picture'
+  field :private, types.Boolean, 'Is team private?' # TODO Rename to is_private
+  field :team_graphql_id, types.String # TODO Why do we need this? Rename to 'team_id'
 
-  field :pusher_channel do
-    type types.String
+  field :trash_count, types.Int, 'Count of items in trash' do # TODO This looks complicated - can it be removed?
+    resolve ->(team, _args, _ctx) {
+      (team.private && (!User.current || (!User.current.is_admin && TeamUser.where(team_id: team.id, user_id: User.current.id).last.nil?))) ? 0 : team.trash_count
+    }
+  end
```

Comment:
Agree to remove, but probably requires work on the frontend - but it should for sure be removed from here because it doesn't make sense to implement access-control logic in `PublicTeam`, it should contain only public information.